### ISSUE TITLE
IDISP003. Ignoring implicit calls of Dispose in current code scope

### ISSUE THREAD

**Issue by kuzyaka** [DotNetAnalyzers/IDisposableAnalyzers#82]
IDISP003 could scan current scope against implicit calls of Dispose within call stack

example:
```cs
System.Threading.Mutex mutex = new Mutex(false, "Mutex");
mutex.ReleaseMutex();
mutex.Close(); // Close() method already calls Dispose method
mutex = null;  // false-positive diagnostic, no need for explicit Dispose() because of Close()
```

**Comment by JohanLarsson** [comment]
For the general case it will be tricky as it requires decompilation of binary references.

Also for the specific case with `Mutex` is the recommended way to call `Close` and not:
```cs 
using (var mutex = new Mutex(false, "Mutex"))
{
    ...
}
```

I'm a bit opinionated in that I prefer `using` here.

**Comment by kuzyaka** [comment]
> For the general case it will be tricky as it requires decompilation of binary references.
OK for binary references, but it could be written in user source code and it is legal usage

**Comment by kuzyaka** [comment]
It seems to be pretty complex feature when inspecting current code scope. 
Perhaps some heuristics could be implemented for binary referenced BCL classes.

**Comment by JohanLarsson** [comment]
Yes we could go that way but only if `Close` is preferred over `Dispose` I think. `using` is the nicest way to use disposables as it is short & clear and works with exceptions.

**Comment by kuzyaka** [comment]
Here is the same nuance, as for https://github.com/DotNetAnalyzers/IDisposableAnalyzers/issues/83

If I have shared IDisposable resource between some methods I cannot apply "using" construct

**Comment by JohanLarsson** [comment]
Closing as by design.
